packages/itwin/ec3-widget/src/components/ExportModal.tsx

```diff
@@ -0,0 +1,206 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import "./ExportModal.scss";
+import React, {
+  useCallback,
+  useEffect,
+  useRef,
+  useState,
+} from "react";
+import { IModelApp } from "@itwin/core-frontend";
+import {
+  Button,
+  Modal,
+  ProgressLinear,
+  ProgressRadial,
+  Text,
+  toaster,
+} from "@itwin/itwinui-react";
+import { useEC3JobsClient } from "./api/context/EC3JobsClientContext";
+import { Link, EC3JobCreate, EC3Job, CarbonUploadState } from "@itwin/insights-client";
+
+interface ExportProps {
+  projectName: string;
+  isOpen: boolean;
+  close: () => void;
+  templateId: string | undefined;
+  token: string | undefined;
+}
+
+const ExportModal = (props: ExportProps) => {
+  const PIN_INTERVAL = 1000;
+  const ec3JobsClient = useEC3JobsClient();
+
+  const [jobStatus, setJobStatus] = useState<CarbonUploadState>();
+  const [jobLink, setJobLink] = useState<Link>();
+
+  const intervalRef = useRef<number>();
+
+  const pinStatus = useCallback(
+    (job: EC3Job) => {
+      const intervalId = window.setInterval(async () => {
+        const token =
+          (await IModelApp.authorizationClient?.getAccessToken()) ?? "";
+        if (job.id && token) {
+          const currentJobStatus =
+            await ec3JobsClient.getEC3JobStatus(token, job.id);
+          if (currentJobStatus.status) {
+            if (
+              currentJobStatus.status === CarbonUploadState.Succeeded
+            ) {
+              setJobLink(currentJobStatus._links.ec3Project);
+            }
+            setJobStatus(currentJobStatus.status);
+          } else {
+            setJobStatus(CarbonUploadState.Failed);
+            toaster.negative("Failed to get job status. 😔");
+          }
+        }
+      }, PIN_INTERVAL);
+      intervalRef.current = intervalId;
+    },
+    [setJobLink, setJobStatus, ec3JobsClient]
+  );
+
+  const runJob = useCallback(
+    async (token: string) => {
+      const accessToken =
+        (await IModelApp.authorizationClient?.getAccessToken()) ?? "";
```

Comment:
See how Grouping & Mapping widget allows users to provide their own `accessToken` getter function. Change here accordingly.